src/ir/item.rs

```diff
@@ -858,7 +870,30 @@ impl Item {
                 }
 
                 true
-            })
+            });
+
+        let ids: Vec<_> = if ctx.options().disable_canonical_naming {
+            let mut ids = Vec::new();
+
+            // If target is anonimous we need find its first named ancestor.
+            if target.is_anon() {
+                while let Some(id) = ids_iter.next() {
+                    ids.push(id);
+
+                    if !ctx.resolve_item(id).is_anon() {
```

Comment:
So this breaks at the first anonymous ancestor, how do we deal with nested anonymous structs? This generates `_bindgen_ty_<n>_bindgen_ty_1`, which is not how everything else is generated, what's the reasoning for that?

Reply:
It breaks on the first named ancestor, not anonymous.

```c++
struct foo {
    struct {
        struct {
        } inner;
    } inner;
};

struct bar {
    struct bar2 {
        struct {
        } inner;
    } inner;
};
```

So for the above we generate:

```
foo
foo__bindgen_ty_1
foo__bindgen_ty_1__bindgen_ty_1

bar
bar2
bar2__bindgen_ty_1
```

The reason I do this is to handle anonymous structs without having conflicts.

If I wasn't adding the first named struct then we would generate `__bindgen_ty_1` two times, one for anon struct under `foo` and one for anon struct under `bar2`, which will result to conflicts.

Reply:
Yeah, fair enough. I thought we didn't have nested `__bindgen_ty_*` but we do seem to have them already, so it seems ok.